### PR TITLE
Create blog post

### PR DESCRIPTION
**Pull request by stephnna**
Hi code reviewer,

I created a blog post app.
First, I initialize it with Git, and setup the Postgres database.
I build the project schema by creating and running migration files.
I included Foreign keys and their corresponding indexes.

Kindly review as necessary
Thanks!

### CODE REVIEW

File: db/migrate/20221020131636_add_user_ref_to_posts.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToPosts < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :posts, :user, null: false, foreign_key: true
```
Comment (DammyShittu) [inline db/migrate/20221020131636_add_user_ref_to_posts.rb:3]: - [x] The post should have an `author_id` column, not `user_id`. And so these references should be `author`, not `user`. You can point the foreign key to the user table. You can look at this link for hint: [https://stackoverflow.com/a/42056089](https://stackoverflow.com/a/42056089)

File: db/migrate/20221020131646_add_user_ref_to_likes.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToLikes < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :likes, :user, null: false, foreign_key: true
```
Comment (DammyShittu) [inline db/migrate/20221020131646_add_user_ref_to_likes.rb:3]: - [x] The likes table should have an `author_id` column, not a `user_id`. And so these references should be `author`, not `user`.

File: db/migrate/20221020131619_add_user_ref_to_comments.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddUserRefToComments < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :comments, :user, null: false, foreign_key: true
```
Comment (DammyShittu) [inline db/migrate/20221020131619_add_user_ref_to_comments.rb:3]: - [x] The comment table should have an `author_id` column, not a `user_id`. And so these references should be `author`, not `user`.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @stephnna,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by stephnna** [comment]
Hi @DammyShittu,
 
I believe that in order to maintain a proper naming convention, It is better we use userId in place of authorId since author class does not exist but the user class does as shown below. Just like we have postId referencing post class. but authorId does not have any author class, it rather points to the user class hence I used userId instead of AuthorId. 

blog_app_erd

File: db/migrate/20221020131646_add_user_ref_to_likes.rb (hunk 2 above)
Comment (stephnna) [reply db/migrate/20221020131646_add_user_ref_to_likes.rb:3]: Thanks!

File: db/migrate/20221020131619_add_user_ref_to_comments.rb (hunk 3 above)
Comment (stephnna) [reply db/migrate/20221020131619_add_user_ref_to_comments.rb:3]: Thanks!

**Review by m05e5 — APPROVED** [review]
Hi @stephnna , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Optional suggestions

- [ ] [ optional ] **you're right about the naming convention, but when working on the backend 
 of a project, you have to follow exactly what's on your ER Diagram because in the future you may be working with a team and that's what everyone will take as a reference**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Bondok6** [comment]
# STATUS: APPROVED :green_circle: :clap: 

## Hi @stephnna 🙋‍♂️, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**